include/oneapi/dpl/pstl/utils_ranges.h

```diff
@@ -344,6 +346,45 @@ struct drop_view_simple
     }
 };
 
+//replicate_start_view_simple inserts replicates of the first element m times, then continues with the range as normal.
+// For counting iterator range {0,1,2,3,4,5,...}, and __replicate_count = 3, the result is {0,0,0,0,1,2,3,4,5,...}
+template <typename _R, typename _Size>
+struct replicate_start_view_simple
+{
+    _R __r;
+    _Size __repl_count;
+
+    replicate_start_view_simple(_R __rng, _Size __replicate_count) : __r(__rng), __repl_count(__replicate_count)
+    {
+        assert(__repl_count >= 0 && __r.size() > 0);
```

Comment:
Why a base range cannot be a trivial size range?
I think in some scenarios it can be valid case...

Reply:
If there is no first element  we should do "nothing"...
For example, we can create a vector with size=0, right? right. Such vector is empty.
And operator[0] is UB (or seg fault system exception).. Probably, a user gets assertion like `(idx >= 0 && idx < size())` in "debug" mode - if the implementation does it.
Also I can define `iota_view(0)` and so on...
So, I cannot see any contradiction in case of `replicate_start_view_simple`  as well.

Reply:
OK.  So if we have a trivial size base range, and we send it to `replicate_start_view_simple(empty_range, 3)`, what should the `size()` of that resulting view be?  `0`?

Reply:
Great, implemented as proposed.